pallets/nfts/src/features/approvals.rs

```diff
@@ -165,10 +179,187 @@ impl<T: Config<I>, I: 'static> Pallet<T, I> {
+
+		CollectionApprovals::<T, I>::try_mutate_exists(
+			(&collection, &owner, &delegate),
+			|maybe_approval| -> DispatchResult {
+				let deposit_required = T::CollectionApprovalDeposit::get();
+				let mut current_deposit =
+					maybe_approval.take().map(|(_, deposit)| deposit).unwrap_or_default();
+				if current_deposit < deposit_required {
+					T::Currency::reserve(&owner, deposit_required - current_deposit)?;
+					current_deposit = deposit_required;
+				}
+				*maybe_approval = Some((deadline, current_deposit));
+				Ok(())
+			},
+		)?;
+
+		Self::deposit_event(Event::TransferApproved {
+			collection,
+			item: None,
+			owner,
+			delegate,
+			deadline,
+		});
+
+		Ok(())
+	}
+
+	/// Cancels the transfer of all `collection` items of `owner` to a `delegate`.
+	///
+	/// This function is used to cancel the approval for the transfer of the collection items of
+	/// `owner` to a `delegate`. After canceling the approval, the deposit is returned to the
+	/// `owner` account and the `ApprovalCancelled` event is emitted.
+	///
+	/// - `owner`: The owner of the collection items.
+	/// - `collection`: The identifier of the collection.
+	/// - `delegate`: The account that had permission to transfer collection items.
+	pub(crate) fn do_cancel_collection_approval(
+		owner: T::AccountId,
+		collection: T::CollectionId,
+		delegate: T::AccountId,
+	) -> DispatchResult {
+		let (_, deposit) = CollectionApprovals::<T, I>::take((&collection, &owner, &delegate))
+			.ok_or(Error::<T, I>::Unapproved)?;
+
+		T::Currency::unreserve(&owner, deposit);
+
+		Self::deposit_event(Event::ApprovalCancelled { collection, owner, item: None, delegate });
+
+		Ok(())
+	}
+
+	/// Clears collection approvals.
+	///
+	/// This function is used to clear `limit` collection approvals for the
+	/// collection items of `owner`. After clearing all approvals, the deposit of each collection
+	/// approval is returned to the `owner` account and the `AllApprovalsCancelled` event is
+	/// emitted.
+	///
+	/// - `owner`: The owner of the collection items.
+	/// - `collection`: The collection ID containing the item.
+	/// - `limit`: The amount of collection approvals that will be cleared.
+	pub(crate) fn do_clear_collection_approvals_limit(
+		owner: T::AccountId,
+		collection: T::CollectionId,
+		limit: u32,
+	) -> DispatchResult {
+		if limit == 0 {
+			return Ok(());
+		}
+		let mut removed_approvals: u32 = 0;
+		let mut deposits: BalanceOf<T, I> = Zero::zero();
+		// Iterate and remove each collection approval, returning the deposit back to the `owner`.
+		for (_, (_, deposit)) in CollectionApprovals::<T, I>::drain_prefix((collection, &owner)) {
+			deposits = deposits.saturating_add(deposit);
+			removed_approvals.saturating_inc();
+			if removed_approvals == limit {
+				break;
+			}
+		}
+
+		T::Currency::unreserve(&owner, deposits);
+		Self::deposit_event(Event::AllApprovalsCancelled { collection, item: None, owner });
```

Comment:
The name of the event here no longer matches the functionality. We dont know that all approvals in their entirety have been cancelled, only that all approveds which were specified (by limit) were cancelled.

I think this can maybe just be addressed with a comment here above the event (e.g. 'all specified approvals were cancelled' and in the event doc comments.

Reply:
Suggested to change this to `ApprovalsCancelled` with the number of approvals removed (could be optional, with None stands for all approvals).

Reply:
You mean `AllApprovalsCancelled`?

Reply:
Resolved by adding a new event `ApprovalsCancelled` emitted when multiple approvals of a collection or item were cancelled.